lib/ecto/schema.ex

```diff
@@ -596,7 +607,8 @@ defmodule Ecto.Schema do
   """
   defmacro has_many(name, queryable, opts \\ []) do
     quote do
-      Ecto.Schema.__has_many__(__MODULE__, unquote(name), unquote(queryable), unquote(opts))
+      queryable = Ecto.Schema.__build_assoc_module__(@association_scope, unquote(queryable))
```

Comment:
That's indeed a good Idea!

I think I've been able to simplify the code quite a bit following it <3